crates/sui-graphql-rpc/src/context_data/pg_backend.rs

```diff
@@ -457,19 +457,54 @@ impl GenericQueryBuilder<Pg> for PgQueryBuilder {
         }
 
         // Filters on the package and/ or module that emitted some event
-        if let Some(p) = filter.emitting_package {
-            query = query.filter(events::dsl::package.eq(p.into_vec()));
+        if let Some(pm) = filter.emitting_module {
+            let parts: Vec<_> = pm.splitn(2, "::").collect();
 
-            if let Some(m) = filter.emitting_module {
-                query = query.filter(events::dsl::module.eq(m));
+            if parts.iter().any(|&part| part.is_empty()) {
+                return Err(DbValidationError::InvalidType(
+                    "Empty strings are not allowed".to_string(),
+                ))?;
+            }
+
+            let p = SuiAddress::from_str(parts[0])
+                .map_err(|e| DbValidationError::InvalidType(e.to_string()))?;
+
+            match parts.len() {
+                1 => {
+                    query = query.filter(events::dsl::package.eq(p.into_vec()));
+                }
+                2 => {
+                    query = query.filter(events::dsl::package.eq(p.into_vec()));
+                    query = query.filter(events::dsl::module.eq(parts[1].to_string()));
+                }
+                _ => {
+                    return Err(Error::Internal(
+                        "Invalid type. Type must have 2 or less parts".to_string(),
```

Comment:
nit: grammar

Also, share the problematic string here as well.

```suggestion
                        "Invalid type. Type must have 2 or fewer parts".to_string(),
```